File: 20200215Python-China/0013. Roman to Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        dict_s={'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        if not s:
            return 0
        rep=dict_s[s[-1]]
        for k in range(len(s)-1):
            if dict_s[s[k]]<dict_s[s[k+1]]:
                rep-=dict_s[s[k]]
            else:
                rep+=dict_s[s[k]]
        return rep

File: archives/C#/0013. Roman to Integer.cs
public class Solution {
    public int RomanToInt(string s) {
        Dictionary<char,int> dictS=new Dictionary<char,int>();
        dictS.Add('I',1);
        dictS.Add('V',5);
        dictS.Add('X',10);
        dictS.Add('L',50);
        dictS.Add('C',100);
        dictS.Add('D',500);
        dictS.Add('M',1000);
        
        int lenS=s.Length;
        if(lenS==0){
            return 0;
        };
        int rep=dictS[s[lenS-1]];
        for(int i=0;i<lenS-1;i++){
            if(dictS[s[i]]<dictS[s[i+1]]){
                rep-=dictS[s[i]];
            }
            else{
                rep+=dictS[s[i]];
            }
        }
        return rep;
    }
}
